File: #04/q07 - alter/alter.js
intercal = (arr1, arr2) => arr1.length == 0 ? arr2 :
    arr2.length == 0? arr1 
    : arr1.shift().toString().split("").map(Number).concat(arr2.shift()).concat(intercal(arr1,arr2))

alter = (x) => intercal(
    [...Array(x).keys()].map(val => val + 1),
    [...Array(x).keys()].map(val => (val + 1) * -1)
)

console.log(alter(2));
console.log(alter(1));
console.log(alter(4));

File: #04/q07 - alter/alter.hs
intercal :: [a] -> [a] -> [a]
intercal [] ys = ys
intercal xs [] = xs
intercal (x:xs) (y:ys) = x : y : intercal xs ys


alter :: (Num b, Enum b) => b -> [b]
alter a = intercal [1..a] (map negate [1..a])


main :: IO ()
main = do
    a <- readLn :: IO Int
    print $ alter a
